### ISSUE TITLE
consensus only node

### ISSUE THREAD

**Issue by vtolstov** [nats-io/nats-streaming-server#805]
does it possible to run on two node nats-streaming and on last another only nats-streaming server without storage, so it forms quorum, but don't have data.
Something like mysql galera replication with galera arbitrator.

**Comment by vtolstov** [comment]
If i create noop store, does it can works for such use case? what stan
client do then it connects to such node?

вт, 16 апр. 2019 г., 18:20 Ivan Kozlovic <notifications@github.com>:

**Comment by kozlovic** [comment]
I don't think you can make one of the node store nothing. I don't think that RAFT would work because you cannot prevent that node from being a leader and regardless of role, a RAFT node may need to retrieve logs for it to work, which if you were to completely make all storage for this node a no-op, I think that would break.
NATS Streaming uses two types of stores in clustered mode: the streaming stores (as in standalone mode) where it stores server info, and messages, and clients and subscriptions state is sort of suppressed since it is not needed because they live in the second type of store: the RAFT stores. RAFT needs a storage and we use `go.etcd.io/bbolt` for that. RAFT provides a storage interface so technically you could implement yours, but again, I don't think it is a good idea to make one of the node in the cluster not "persist" anything (as in not be able to retrieve logs). You could make it a memory store maybe. Actually I believe that hashicorp RAFT (the one we use) implement a basic memory store for tests.

**Comment by vtolstov** [comment]
Thank you, my main idea that i have two servers that have storage and it
also have mysql multimaster and runs my app, and i have backup node that
have does not have much resources and i dont want that it handle
workload,but i want to have quorum 2 for raft.

вт, 16 апр. 2019 г., 20:37 Ivan Kozlovic <notifications@github.com>:
